src/client/Microsoft.Identity.Client/ApiConfig/BaseAbstractAcquireTokenParameterBuilder.cs

```diff
@@ -84,6 +84,11 @@ protected virtual void Validate()
 
         internal void ValidateAndCalculateApiId()
         {
+            if (CommonParameters.AuthorityOverride?.AuthorityType == AuthorityType.Ciam)
```

Comment:
Why do we need this? Adds complexity to the authority transform logic?

Reply:
It is actually intended to reduce complexity in developer scenarios. Rather than allowing the CIAM authorities to be set at the request level and introduce bugs like we have seen in the past. We decided to only allow CIAM authorities to be set at the app level.